### PR TITLE
FIXIT: Limit preload links to 20.

### LINKED ISSUE #59: Limit to 20 preloads
The Google SXG cache has a [limit of 20 preloads](https://github.com/google/webpackager/blob/master/docs/cache_requirements.md#user-content-google-sxg-cache:~:text=There%20may%20be%20no%20more%20than%2020%20rel%3Dpreloads.). I don't see anything in the webpackager code that guarantees that requirement is met. (Did I miss it?) We should provide a [default enabled] option to do so.

Seems especially necessary given the optional HTMLTask to [promote preload tags to headers](https://github.com/google/webpackager/blob/master/processor/htmlproc/htmltask/preload.go), where the former are more prominent today, with no such limit.

### PR DESCRIPTION
**Pull request by banaag**
Fixed #59

### CODE REVIEW

File: processor/commonproc/preload.go [hunk 1]
```diff
@@ -41,16 +44,19 @@ func (*extractPreloadHeaders) Process(resp *exchange.Response) error {
...
+	numPreloads := 0
 	for _, value := range values {
 		links, err := httplink.Parse(value)
 		if err != nil {
 			log.Printf("warning: %v -- this header was ignored", err)
 			continue
 		}
 		for _, link := range links {
-			if link.IsPreload() {
+			if numPreloads < maxNumPreloads && link.IsPreload() {
```
Comment (twifkak) [inline processor/commonproc/preload.go:55]: I think you want this condition to be nested; the else fallback is not desired. (In the test, 21 should not get a link header at all.)

**Review by twifkak — CHANGES_REQUESTED** [review]

File: processor/commonproc/preload.go [hunk 2]
```diff
@@ -32,6 +32,9 @@ const headerKey = "Link"
 // the Link HTTP headers when the response is turned into a signed exchange.
 var ExtractPreloadHeaders processor.Processor = &extractPreloadHeaders{}
 
+// maxNumPreloads is the maximum number of preload links allowed by WebPackager.
```
[the remark this answers is not in the document]
Comment (banaag) [reply processor/commonproc/preload.go:35]: Done.

File: processor/commonproc/preload.go (hunk 1 above)
Comment (banaag) [reply processor/commonproc/preload.go:55]: Done.

File: processor/commonproc/preload_test.go [hunk 3]
```diff
@@ -127,6 +127,65 @@ func TestExtractPreloadHeaders(t *testing.T) {
...
+				"Link: <https://example.com/photo13.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo14.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo15.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo16.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo17.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo18.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo19.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo20.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/photo21.jpg>;rel=\"preload\";as=\"image\"\r\n",
+				"Link: <https://example.com/>;rel=\"start\"\r\n",
```
Comment (twifkak) [inline processor/commonproc/preload_test.go:156]: If you're looking for another fixit task after this PR, we should also strip this. :)

https://github.com/google/webpackager/blob/master/docs/cache_requirements.md#user-content-google-sxg-cache:~:text=All%20rel%20parameters%20must%20be%20either%20preload%20or%20allowed%2Dalt%2Dsxg.

**Review by twifkak — APPROVED** [review]

**Pull request merged**